pipeline-runner/init.R

```diff
@@ -350,11 +389,23 @@ init <- function() {
         flog.appender(appender.tee(file.path(dump_folder, "logs.txt")))
 
         tryCatchLog({
+
+                # start heartbeat as a different process in the background
+                message("Starting heartbeat")
+                # message inside r_bg will ONLY be printed into /tmp/[out|err]
+                # to see them
+                # 1. log into the R container
+                # 2. cat /tmp/out  o tail -f /tmp/out
+                r_bg(func=start_heartbeat, args=list(
```

Comment:
`r_bg` actually returns a [process object](https://callr.r-lib.org/reference/r_bg.html#value) which has a [kill function ](https://www.tidyverse.org/blog/2018/09/processx-3.2.0/). Maybe you can use this function to kill the heartbeat process at the end of task completion?

Reply:
sounds good will do